### DISCUSSION TITLE
speaker diarization and whisper segment

### DISCUSSION THREAD

**Discussion by Atefeh197** [openai/whisper#1255] [General]
Hi

I want to do speaker diarization on whisper's output.
I know Whisper generates the segments for each audio based on this result:
'segments': [{'seek': 0.0, 'start': 0.46, 'end': 1.98, 'text': ' Hi, how are you', ......

My method is that extract embeddings for each segment and then use a diarizing model for labeling.

I would like to know if the segments that are extracted by whisper are based on speaker change detection or something like that.
I mean  Is there only one speaker speaking in each segment?

**Comment by HeadStudios** [comment]
Following.

**Reply by Majdoddin** [reply]
@Atefeh197 @HeadStudios 
[www.lexicaps.com](http://www.lexicaps.com/) seamlessly adds diarization to Whispers transcription. No 3rd party packages.
Announcement: https://github.com/openai/whisper/discussions/1537
Repo: https://github.com/Majdoddin/lexicaps
